### PR TITLE
ADE-632: Add default schemas

### PR DESCRIPTION
**Pull request by shouvSarker**

### CODE REVIEW

File: backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
+package au.com.agiledigital.structured_form.helpers;
+
+/**
+ * Supplies default schemas when there is no schema saved in the database.
+ */
+public class SchemaHelper {
```
Comment (dspasojevic) [inline backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java]: Prefer that this is named `DefaultSchema`.

File: backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+  /**
+   * The default page schema, this will show up when trying to create a page
+   * from this plugin without setting a custom schema.
+   * Currently it is from markdown ADR: https://adr.github.io/madr/#the-template.
+   */
+  public static final String defaultSchema = "{\n"
```
Comment (dspasojevic) [inline backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java]: Prefer that this is named `schema` when the class is renamed.

File: backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    + "        \"type\": \"string\"\n"
+    + "      }\n"
+    + "    }\n"
+    + "  }\n"
+    + "}";
+
+  /**
+   * Default schema for ui styling.
+   */
+  public static final String defaultUiSchema = "{\n"
```
Comment (dspasojevic) [inline backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java]: Prefer that this is renamed `uiSchema` when the class is renamed.

File: backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    + "  },\n"
+    + "  \"problemStatement\": {\n"
+    + "    \"ui:widget\": \"textarea\"\n"
+    + "  }\n"
+    + "}";
+
+  /**
+   * Default index fields for ADR schema.
+   */
+  public static final String defaultIndex = "{\n"
```
Comment (dspasojevic) [inline backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java]: Prefer that his is renamed `indexFields` (or is this also a schema?) when the class is renamed.

File: backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java (hunk 1 above)
Comment (shouvSarker) [reply backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java]: Renamed.

File: backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java (hunk 2 above)
Comment (shouvSarker) [reply backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java]: Renamed

File: backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java (hunk 4 above)
Comment (shouvSarker) [reply backend/src/main/java/au/com/agiledigital/structured_form/helpers/SchemaHelper.java]: In it's current set up, it is also a schema. Renamed.

File: backend/src/main/java/au/com/agiledigital/structured_form/helpers/DefaultSchema.java [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+ * Supplies default schemas when there is no schema saved in the database.
+ */
+public class DefaultSchema {
+
+  /**
+   * The default page schema, this will show up when trying to create a page
+   * from this plugin without setting a custom schema.
+   * Currently it is from markdown ADR: https://adr.github.io/madr/#the-template.
+   */
+  public static final String schema = "{\n"
```
Comment (dspasojevic) [inline backend/src/main/java/au/com/agiledigital/structured_form/helpers/DefaultSchema.java]: Pity that you can't use multiline literals and still target JRE 8.

Comment (shouvSarker) [reply backend/src/main/java/au/com/agiledigital/structured_form/helpers/DefaultSchema.java]: There were complains even before java 8 came out!

https://blog.joda.org/2008/01/java-7-multi-line-string-literals_594.html

**Comment by shouvSarker** [comment]
Code smells have been removed. NOSONAR comment has been used on specific lines to ignore issues rising from duplicate string literals while declaring default schemas.

File: backend/src/main/java/au/com/agiledigital/structured_form/helpers/DefaultSchema.java [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  public static final String INDEX_SCHEMA = "{\n"
+    + "  \"index\": [\n"
+    + "    {\n" // NOSONAR string schema declaration can contain duplicates
+    + "      \"key\": \"status\",\n"
+    + "      \"index\": 1,\n"
+    + "      \"type\": \"string\"\n" // NOSONAR string schema declaration can contain duplicates
+    + "    },\n"
+    + "    {\n"
+    + "      \"key\": \"date\",\n"
+    + "      \"index\": 1,\n"
```
Comment (robinMcA) [inline backend/src/main/java/au/com/agiledigital/structured_form/helpers/DefaultSchema.java]: two items have index 1

Comment (shouvSarker) [reply backend/src/main/java/au/com/agiledigital/structured_form/helpers/DefaultSchema.java]: Thanks for catching this, updated!

**Pull request merged**